backend/src/main/kotlin/app/ehrenamtskarte/backend/regions/database/Setup.kt

```diff
@@ -1,11 +1,418 @@
+    ),
+    listOf(
+        "Landkreis",
+        "Würzburg",
+        "09679",
+        "http://www.landkreis-wuerzburg.de/Politik_Behörde/Servicestelle_Ehrenamt/Bayerische_Ehrenamtskarte/"
+    ),
+    listOf("Landkreis", "Weißenburg-Gunzenhausen", "09577", "http://www.altmuehlfranken.de/ehrenamtskarte"),
+    listOf("Stadt", "Weiden i.d.OPf.", "09363", "https://www.weiden.de/kultur/ehrenamt/ehrenamtskarte"),
+    listOf(
+        "Landkreis",
+        "Weilheim-Schongau",
+        "09190",
+        "https://www.weilheim-schongau.de/buergerservice/ehrenamtskarte/"
+    ),
+    listOf("Landkreis", "Wunsiedel i.Fichtelgebirge", "09479", "http://www.landkreis-wunsiedel.de/ehrenamtskarte")
+)
+
+fun setupDatabase() {
     SchemaUtils.create(
         Regions
     )
 
-    executeScript("sql/create_regions.sql")
+    transaction {
+        val projects = ProjectEntity.all()
+        val dbRegions = RegionEntity.all()
+
+        // Delete old regions in database
+        dbRegions.forEach { dbRegion ->
+            val isEakRegion = eakRegions.any { it[2] == dbRegion.regionIdentifier }
+            val isProjectRegion = projects.any { it.regionIdentifier() == dbRegion.regionIdentifier }
+            if (!isEakRegion && !isProjectRegion) {
+                dbRegion.delete()
+            }
+        }
+
+        // Create a mock region for each project to save the stores to
+        projects.forEach { project ->
+            // Project ids and therefore the mock regions should never change
+            // Therefore we only need to create the mock region if not already existing
+            if (dbRegions.none { it.regionIdentifier == project.regionIdentifier() }) {
+                RegionEntity.new {
+                    projectId = project.id
+                    name = project.project
+                    prefix = ""
+                    regionIdentifier = project.regionIdentifier()
+                    website = ""
+                }
+            }
+        }
+
+        // Create or update eak regions in database
+        val eakProject = projects.first { it.project == "bayern.ehrenamtskarte.app" }
+        eakRegions.forEach { eakRegion ->
+            val dbRegion = dbRegions.find { it.regionIdentifier == eakRegion[2] }
+            if (dbRegion == null) {
+                RegionEntity.new {
+                    projectId = eakProject.id
+                    name = eakRegion[1]
+                    prefix = eakRegion[0]
+                    regionIdentifier = eakRegion[2]
+                    website = eakRegion[3]
+                }
+            } else {
+                dbRegion.name = eakRegion[1]
+                dbRegion.prefix = eakRegion[0]
+                dbRegion.website = eakRegion[3]
+            }
+        }
+    }
+}
+
+private fun ProjectEntity.regionIdentifier(): String {
+    return project.hashCode().toString().slice(0..4)
```

Comment:
any better ideas on how to create a unique 5 digit id?

Reply:
It has to be unique (so null, undefined, mock) is not possible for both. The project id is not 5 chars long.

Reply:
I think we should probably make the regionIdentifier nullable and use null. A dummy region does not have a kreisschlüssel nevertheless :) (also for other projects this kreisschlüssel semantic might be too specific anyways...)

(
In a unique, but nullable column, null values can appear multiple times, they are not subject to the uniqueness constraint: From https://www.postgresql.org/docs/current/ddl-constraints.html#DDL-CONSTRAINTS-UNIQUE-CONSTRAINTS

> In general, a unique constraint is violated if there is more than one row in the table where the values of all of the columns included in the constraint are equal. However, two null values are never considered equal in this comparison. That means even in the presence of a unique constraint it is possible to store duplicate rows that contain a null value in at least one of the constrained columns. This behavior conforms to the SQL standard, but we have heard that other SQL databases might not follow this rule. So be careful when developing applications that are intended to be portable.

)

Reply:
Done